internal/agent/device/applications/provider/provider.go

```diff
@@ -875,6 +875,10 @@ func ensureQuadlet(readWriter fileio.ReadWriter, appPath string) error {
 		return fmt.Errorf("parsing quadlet spec: %w", err)
 	}
 
+	if errs := validation.ValidateQuadletNames(spec); len(errs) > 0 {
```

Comment:
minor, but lets move this down like ValidatingCrossReferences so that we can gather all of the errors.